### PR TITLE
Remove unavailable commands

### LINKED ISSUE #43: Manpage lists lots of unavailable commands
Did not count them but it feels as if 3/4 of the commands mentioned in the manpage are not available...

```bash
mihi@da1:~$ for cmd in $(man hashbang | perl -ne '/^       ([^ ]+) - / && print "$1 "'); do echo -n $cmd:; which $cmd || echo '-'; done
#!:-
perl:/usr/bin/perl
python:/usr/bin/python
ruby:/usr/bin/ruby
haskell:-
lua:/usr/bin/lua
clojure:-
go:-
nodejs:/usr/bin/nodejs
sbcl:-
ghc:-
gcc:/usr/bin/gcc
spidermonkey:-
vim:/usr/bin/vim
nano:/usr/bin/nano
joe:-
pico:/usr/bin/pico
e3:-
ne:-
mcedit:/usr/bin/mcedit
zile:-
mg:-
jed:-
pass:-
encfs:-
gpg:/usr/bin/gpg
md5sum:/usr/bin/md5sum
shasum:/usr/bin/shasum
bcrypt:-
ccrypt:-
calcurse:-
calendar:/usr/bin/calendar
remind:-
wyrd:-
taskwarrior:-
tudu:-
zsh:/usr/bin/zsh
fish:-
ksh:-
mutt:/usr/bin/mutt
alpine:-
slrn:-
offlineimap:-
mu:-
ledger:-
units:-
dc:-
qalc:-
bc:-
weechat-curses:/usr/bin/weechat-curses
irssi:/usr/bin/irssi
centerim:-
finch:-
bitlbee:-
barnowl:-
pork:-
scrollz:-
ii:-
sic:-
erc:-
newsbeuter:-
rsstail:-
canto:-
links:-
elinks:/usr/bin/elinks
lynx:/usr/bin/lynx
w3m:-
html2text:-
sqlite3:-
mariadb:-
postgresql:-
mongodb:-
redis:-
mc:/usr/bin/mc
scp:/usr/bin/scp
rsync:/usr/bin/rsync
rsynccrypto:-
duplicity:-
vifm:-
ranger:-
du:/usr/bin/du
ncdu:/usr/bin/ncdu
stow:/usr/bin/stow
find:/usr/bin/find
locate:/usr/bin/locate
tree:-
atool:-
zip:/usr/bin/zip
unzip:/usr/bin/unzip
p7zip:-
tar:/bin/tar
dar:-
gzip:/bin/gzip
zpaq:-
iptraf:-
nethogs:-
slurm:-
nmap:-
ngrep:-
tcpdump:-
trickle:-
ifstat:-
iftop:-
mtr:-
telnet:/usr/bin/telnet
netpipes:-
ssh:/usr/bin/ssh
siege:-
lftp:-
curl:/usr/bin/curl
rtorrent:-
aria2:-
ipcalc:-
socat:-
netcat:/bin/netcat
ssh-copy-id:/usr/bin/ssh-copy-id
zeromq:-
rabbitmq:-
optipng:-
imagemagick:-
gd:-
cvs:-
svn:-
mercurial:-
git:/usr/bin/git
tig:/usr/bin/tig
cloc:-
diff:/usr/bin/diff
vimdiff:/usr/bin/vimdiff
vimpager:-
hub:-
ctags:/usr/bin/ctags
cmake:/usr/bin/cmake
zangband:-
nethack:-
slashem:-
cmatrix:-
frotz:-
bsdgames:-
bb:-
sl:-
bastet:-
greed:-
gnugo:-
gnuchess:-
moon-buggy:-
typespeed:-
tinyfugue:-
tintin++:-
kbtin:-
pennmush:-
bonnie++:-
htop:/usr/bin/htop
dstat:-
iotop:-
sysdig:-
strace:/usr/bin/strace
cpulimit:-
cgroups:-
command-not-found:-
tmux:/usr/bin/tmux
screen:/usr/bin/screen
dtach:-
pv:-
ttyrec:-
tsung:-
xargs:/usr/bin/xargs
parallel:-
tpp:-
ack:-
ag:-
repl:-
watch:/usr/bin/watch
libev:-
libevent:-
lame:-
cowsay:-
dos2unix:-
unix2dos:-
2004:-
2006:-
2010:-
2014:-
```

If that is intentional (of course the year numbers are not supposed to be commands but most of the rest) I can create a pull request that removes all of them from the manpage (after manual checking)

### PR DESCRIPTION
**Pull request by schierlm**
Closes #43

### CODE REVIEW

**Comment by ChickenNuggers** [comment]
Did you make sure that this list is compatible with https://github.com/hashbang/shell-etc/issues/43#issuecomment-168362794 ?

**Comment by schierlm** [comment]
I re-checked available binaries after he made that comment, and tried not to remove those that got installed meanwhile

File: man/man7/hashbang.7 [hunk 1]
```diff
@@ -56,24 +53,15 @@ applications written in javascript.
 sbcl - (Steel Bank Common Lisp) A Lisp implementation that features a high
 performance native compiler, Unicode support, and threading.
 
-ghc - (The Glorious Glasgow Haskell Compilation System) a native code compiler
```
Comment (daurnimator) [inline man/man7/hashbang.7]: ghc is installed.

File: man/man7/hashbang.7 [hunk 2]
```diff
@@ -38,9 +38,6 @@ code readability.
 
 ruby - A dynamic, object-oriented general-purpose programming language.
 
-haskell - A standardized, general-purpose programming language with non-strict
```
Comment (daurnimator) [inline man/man7/hashbang.7]: ghc is available.

File: man/man7/hashbang.7 [hunk 3]
```diff
@@ -83,51 +71,21 @@ pico - (Pine Composer) a text editor originally integrated with the pine e-mail
...
-using an arbitrary directory as storage for the encrypted files.
-
 gpg - (GNU Privacy Guard) A GPL Licensed alternative to the PGP suite of
 cryptographic software compliant with RFC 4880.
 
 md5sum - Calculates and verifies 128-bit MD5 hashes as described in RFC 1321.
 
 shasum - Calculates and verifies SHA hashes.
-
-bcrypt - A key derivation function for passwords based on the Blowfish cipher.
```
Comment (daurnimator) [inline man/man7/hashbang.7]: hrm, I thought I installed this.

File: man/man7/hashbang.7 [hunk 4]
```diff
@@ -144,112 +102,22 @@ includes many features of the C shell.
...
 irssi - A text-based IRC client written in the C programming language.
-
-centerim - A text based menu and window-driven instant messaging interface that
-supports ICQ, Yahoo!, AIM, MSN, IRC, XMPP, LiveJournal, and Gadu-Gadu
-protocols.
-
-finch - A console-based instant messaging client based on the libpurple
-library.
-
-bitlbee - Bitlbee brings Instant Messaging to IRC clients. It has support for
```
Comment (daurnimator) [inline man/man7/hashbang.7]: bitlbee is installed but not running.

File: man/man7/hashbang.7 [hunk 5]
```diff
@@ -144,112 +102,22 @@ includes many features of the C shell.
...
-rendering and background downloading.
-
 elinks - Similar to links, but also supports Form Input, Password Management,
 and Tabbed Browsing
 
 lynx - A general purpose distributed information web browser.
 
 w3m - A text based web browser and pager.
-
-html2text - Reads an HTML document and outputs plain text characters.
```
Comment (daurnimator) [inline man/man7/hashbang.7]: I thought I installed this.

File: man/man7/hashbang.7 [hunk 6]
```diff
@@ -144,112 +102,22 @@ includes many features of the C shell.
...
-mariadb - A fork of the MySQL relational database management system led by the
-original developer of MySQL.
-
-postgresql - An Object-relational database management system that emphasises
-extensibility and standards-compliance.
-
-mongodb - A document-oriented NoSQL database that uses JSON-like documents with
-dynamic schemas.
-
-redis - A networked, in-memory, key-value data store with optional durability
```
Comment (daurnimator) [inline man/man7/hashbang.7]: redis is installed.

File: man/man7/hashbang.7 [hunk 7]
```diff
@@ -283,94 +141,28 @@ locate - Searches a prebuilt database for files on a filesystem.
...
 FTPS, SCP, SFTP, TFTP, DICT, TELNET, LDAP or FILE).
 
-rtorrent - A BitTorrent client for ncurses using the libtorrent library.
-
-aria2 - A utility for downloading files via HTTP(S), FTP, BitTorrent, and
-Metalink.
-
-ipcalc - A program that calculates IP information for a host.
-
-socat - (SOcket CAT) A command line based utility that establishes two
```
Comment (daurnimator) [inline man/man7/hashbang.7]: I thought I installed this too....

File: man/man7/hashbang.7 [hunk 8]
```diff
@@ -378,7 +170,7 @@ client-server architecture.
 svn - (Subversion) A software versioning and revision control system
 maintained by apache and designed as a successor to CVS
 
-mercurial - A distributed revision control system designed for high
```
Comment (daurnimator) [inline man/man7/hashbang.7]: why the rename?

File: man/man7/hashbang.7 [hunk 9]
```diff
@@ -386,16 +178,10 @@ integrity, and support for distributed, non-linear workflows.
 
 tig - A text-mode interface for git.
 
-cloc - Counts and computes differences of lines of source code and comments.
```
Comment (daurnimator) [inline man/man7/hashbang.7]: sigh.. (thought it was installed)

File: man/man7/hashbang.7 [hunk 10]
```diff
@@ -386,16 +178,10 @@ integrity, and support for distributed, non-linear workflows.
...
-
 diff - Compares files line by line.
 
 vimdiff - Edits 2 - 4 versions of a file with vim while showing differences.
 
-vimpager - Allows vim to be used as a pager for things like man pages.
-
-hub - Wrapper around git(1) that allows better integration with Github.
```
Comment (daurnimator) [inline man/man7/hashbang.7]: we should install this.

File: man/man7/hashbang.7 [hunk 11]
```diff
@@ -472,51 +212,14 @@ user to access multiple separate terminal sessions inside a single terminal.
...
 user to access multiple separate terminal sessions in a single terminal.
 
-dtach - A program that emulates the detach feature of screen.
-
-byobu - An enhancement for the terminal multiplexers Screen or Tmux that can be
-used to provide on screen notification or status as well as tabbed multi-window
-management.
-
 .SS Misc. / Unsorted (Sort these!)
-pv - Monitors the progress of data through a pipe.
```
Comment (daurnimator) [inline man/man7/hashbang.7]: thought this was installed.... apparnelty not. lets install.

File: man/man7/hashbang.7 [hunk 12]
```diff
@@ -472,51 +212,14 @@ user to access multiple separate terminal sessions inside a single terminal.
...
-ack - A grep-like text finder.
-
-ag - A significantly faster replacement to ACK with a built in VCS.
-
-repl - Wraps a non-interactive command in an interactive read-eval-print-loop
-prompt.
-
 watch - Executes a program periodically, showing the output fullscreen.
-
-libev - A high-performance event loop for C.
```
Comment (daurnimator) [inline man/man7/hashbang.7]: installed.

File: man/man7/hashbang.7 [hunk 13]
```diff
@@ -472,51 +212,14 @@ user to access multiple separate terminal sessions inside a single terminal.
...
-
-libevent - Provides a mechanism to execute a callback function when a specific
```
Comment (daurnimator) [inline man/man7/hashbang.7]: installed.

File: man/man7/hashbang.7 [hunk 14]
```diff
@@ -472,51 +212,14 @@ user to access multiple separate terminal sessions inside a single terminal.
...
-event occurs on a file descriptor or after a timeout has been reached.
-
-lame - An MP3 encoder.
-
-cowsay - Generates ASCII pictures of a cow with a message.
-
-dos2unix - Converts line breaks in a text file from DOS format to Unix format.
```
Comment (daurnimator) [inline man/man7/hashbang.7]: thought I installed this....

File: man/man7/hashbang.7 (hunk 2 above)
Comment (schierlm) [inline man/man7/hashbang.7]:
```
mihi@da1:~$ ghc
Could not find the database of available applications, run update-command-not-found as root to fix this
ghc: command not found
```
According to https://packages.debian.org/stable/i386/ghc/filelist, the binary should be called `/usr/bin/ghc`

File: man/man7/hashbang.7 (hunk 6 above)
Comment (schierlm) [inline man/man7/hashbang.7]: how to launch?

File: man/man7/hashbang.7 (hunk 8 above)
Comment (schierlm) [inline man/man7/hashbang.7]: It seemed to me that the manpage tries to write the programs like you invoke them. Mercurial is invoked as `hg` and there is no symlink. If that is not the case, I can add some comment that it is invoked as `hg` :)

File: man/man7/hashbang.7 (hunk 6 above)
Comment (daurnimator) [inline man/man7/hashbang.7]: server or client or sentinel?
but, e.g. `redis-server`

**Comment by schierlm** [comment]
Tell me when you believe you are finished installing and I'll update the pull request :)

File: man/man7/hashbang.7 (hunk 6 above)
Comment (schierlm) [inline man/man7/hashbang.7]: you win. I think I tried to tab-complete `redis` and it found nothing, but now it does indeed find them :)

**Comment by schierlm** [comment]
Or I'll just close it.

**Comment by KellerFuchs** [comment]
@schierlm I think I fixed today the install issues @daurnimator had, so this should finally get handled.

File: man/man7/hashbang.7 (hunk 4 above)
Comment (KellerFuchs) [inline man/man7/hashbang.7]: Couldn't find a package for `centerim`.

**Comment by KellerFuchs** [comment]
Going towards solving this definitely: I solved the “disappearing packages” issue, and I (re)installed the following packages:

```
aria2
bcrypt
bitlbee
cloc
clojure1.6
cowsay
dos2unix
finch
ghc
golang-go
html2text
imagemagick
ipcalc
pass
pv
rtorrent
socat
```

**Comment by schierlm** [comment]
Ok, went over the diff again and updated it

- Added back what got installed
- If the binary and the program name differ, add the binary in square brackets

I hope I did not miss anything.

**Comment by KellerFuchs** [comment]
Ok, installed the following additional packages which seem universally useful:
```
encfs remind wyrd tudu dc qalc duplicity ranger atool p7zip zpaq nmap mtr iperf lftp zangband
 slashem frotz byobu tsung silversearcher-ag{,-el}
```

`silversearcher-ag{,-el}` are `ag` and it's Emacs integration.

**Comment by KellerFuchs** [comment]
Aside from that, I checked your diff and it seems ok.
I will happily merge this once you add the needed packages.

**Comment by schierlm** [comment]
Added everything except `iperf` which hasn't been on the previous manpage either.

**Pull request closed without merging**